### ISSUE TITLE
Support check for "redundant enclosing brace" in string interpolation.

### ISSUE THREAD

**Issue by Synesso** [olafurpg/scalafmt#690]
## Enhancement request

Prevent redundant enclosing block in string interpolation, as detected by IntelliJ's default code style. 

```scala
s"Hello ${"dlroW".reverse}"
```
is OK

But
```scala
val s = "World"
s"Hello ${s}"
```
is reformatted to
```
val s = "World"
s"Hello $s"
```

**Comment by ysusuk** [comment]
does it make sense to wait until https://github.com/scalameta/scalameta/issues/251 will be implemented?

**Comment by olafurpg** [comment]
No need to wait for that issue, scalafmt does not use quasi quotes. You can find the expression inside an interpolation string by matching against Interpolator.Part iirc

**Comment by ysusuk** [comment]
Hi @olafurpg! shall it be another rewrite rule?

**Comment by olafurpg** [comment]
I was thinking of bundling it together with RedundantBraces but disabled
behind a setting. We can enable the setting by default in the next minor
bump. Wdyt?

**Comment by ysusuk** [comment]
i put in https://github.com/olafurpg/scalafmt/blob/master/core/src/main/scala/org/scalafmt/rewrite/RedundantBraces.scala#L56 this cod

```scala
case Pat.Interpolate(x, y, z) =>
        println("!!! int pat")
      case Term.Interpolate(x, y, z) =>
        println("!!! int term")
```

**Comment by olafurpg** [comment]
That's weird,

```scala
@ q""" 1 + q"a $${b}" """.collect {
    case Term.Interpolate(a, b, c) =>
      a -> b -> c
  }
res7: List[((Term.Name, collection.immutable.Seq[Lit]), collection.immutable.Seq[Term])] = List(((q, List("a ", "")), List(b)))
```

**Comment by ysusuk** [comment]
looks like some strange behaviour in tests (e.g. https://github.com/olafurpg/scalafmt/issues/657)

changed

```scala
s"Hello ${c}"
```
to
```scala
object a {
  def b(c: String) = s"Hello ${c}"
}
```
and it works

**Comment by olafurpg** [comment]
This issue was fixed by @ysusuk in #703
